packages/plugin-rust-cargo/index.js

```diff
@@ -0,0 +1,45 @@
+import liveResolverQuery from '@octolinker/resolver-live-query';
+import * as TOML from '@iarna/toml';
+import insertLink from '@octolinker/helper-insert-link';
+import { tomlRegExKeyValue } from '@octolinker/helper-regex-builder';
+
+function linkDependency(blob, key, value) {
+  const regex = tomlRegExKeyValue(key, value);
+  return insertLink(blob, regex, this);
+}
+
+function processTOML(blob, plugin, config) {
+  let results = [];
+  const json = TOML.parse(blob.toString());
+  const toml = TOML.stringify(json.dependencies).split('\n');
+
+  toml.forEach(line => {
+    const match = /^([^=\s]+)\s*=\s*"([^"\s]+)"$/.exec(line);
+
+    if (match) {
+      const linked = linkDependency.call(plugin, blob, match[1], match[2]);
+
+      results = results.concat(linked);
+    }
+  });
+
+  return results;
+}
+
+export default {
+  name: 'Rust Cargo',
+
```

Comment:
By adding `needsContext: true`, the [blob reader](https://github.com/OctoLinker/OctoLinker/blob/9e1caa3ed26514cb38c1c7b6a9f5fc5a7ec0195e/packages/core/app.js#L25) will fetch the full source code when the plugin is invoked on a PR split view. Otherwise the `TOML.parse` will fail if the source is divided and therfore not valid, similar to [plugin-npm-manifest](https://github.com/OctoLinker/OctoLinker/blob/49be306c492cae96930cfa16d163dc31076ecc60/packages/plugin-npm-manifest/index.js#L34) where we need a valid json string.